src/index.d.ts

```diff
@@ -115,6 +139,7 @@ declare module 'bootstrap-vue' {
   export interface Progress extends Vue {}
   export interface Table extends Vue {
     refresh: () => void
+    fields: TableFieldObject | TableFieldArray
```

Comment:
Maybe: `fields: TableFieldObject | TableFieldArray | string` ?

Although the fields definition can be a mix of string entries and object entries... does the `TableFieldArray` type handle mixed value arrays (i.e. strings and objects)?

Reply:
Also, we discourage the use of an object being used to describe the fields, as field order cannot be guaranteed, and may cause SSR rehydration issues on some browsers.

Reply:
Fields can also be `null`, which means that `b-table` will sample the first row of data to generate the field names.

So maybe it should be `fields?: TableFieldObject | TableFieldArray | string[]` ?

Reply:
The way I do it is:
```ts
type TableFieldObject = { key: string } | string; // expand out the object - just here for example
type TableFieldArray = Array<TableFieldObject>;
```

Reply:
`TableFieldArray` is defined as `Array<string | {key: string & TableField}>` to handle arrays of strings or objects (since, as you said, they can be mixed).

If you're discouraging objects being used, I'm willing to remove it, but then it comes down to a question of are we typing all supported options, or just the ones we prefer?

I will also add handling for `fields` to be `null`.